quixstreams/models/topics/admin.py

```diff
@@ -0,0 +1,194 @@
+        :return: a dict with topic names and their respective `TopicConfig`
+        """
+        futures_dict = {}
+        cluster_topics = self.list_topics()
+        if existing_topics := [
+            topic for topic in topic_names if topic in cluster_topics
+        ]:
+            futures_dict = self._admin_client.describe_configs(
+                [confluent_topic_config(topic) for topic in existing_topics]
+            )
+        configs = {
+            config_resource.name: {c.name: c.value for c in config.result().values()}
+            for config_resource, config in futures_dict.items()
+        }
+        return {
+            topic: TopicConfig(
+                num_partitions=len(cluster_topics[topic].partitions),
+                replication_factor=len(cluster_topics[topic].partitions[0].replicas),
+                extra_config=configs[topic],
+            )
+            if topic in existing_topics
+            else None
+            for topic in topic_names
+        }
+
+    def _finalize_create(self, futures: Dict[str, Future], timeout: int):
+        """
+        The confirmation step for topic creation.
+
+        :param futures: a dict of futures as generated by Confluent's
+            `AdminClient.create_topics()`
+        :param timeout: How long to confirm topics before raising exception
+        """
+        exceptions = {}
+        successful = []
+        existed = []
+        stop_time = time.time() + timeout
+        while futures and time.time() < stop_time:
+            time.sleep(1)
+            for topic_name in list(futures.keys()):
+                future = futures[topic_name]
+                if future.done():
+                    try:
+                        future.result()
+                        successful.append(topic_name)
+                    except KafkaException as e:
+                        # Topic was maybe created by another instance
+                        if e.args[0].name() == "TOPIC_ALREADY_EXISTS":
+                            existed.append(topic_name)
+                        else:
+                            exceptions[topic_name] = e.args[0].str()
+                    # Not sure how these get raised, but they are supposedly possible
+                    except (TypeError, ValueError) as e:
+                        exceptions[topic_name] = e
+                    del futures[topic_name]
+        if successful:
+            logger.info(f"Successfully created topics: {successful}")
+        if existed:
+            logger.info(f"These topics already exist: {existed}")
+        if exceptions:
+            raise CreateTopicFailure(exceptions)
+        if futures:
+            raise CreateTopicTimeout(
+                f"Timed out waiting for creation status for topics:\n"
+                f"{pprint.pformat([topic_name for topic_name in futures])}"
+            )
+
+    def create_topics(
+        self, topics: List[Topic], timeout: int = 10, finalize_timeout: int = 20
```

Comment:
Let's increase the default value of `finalize_timeout` to 60.

"20" seems to be too close to "10" in case cluster doesn't respond fast enough and there're multiple topics